### PR TITLE
Issue 10

### PR DESCRIPTION
**Pull request by IvaniGabrovsky**
I Added support for configuration file (-c --config).

### CODE REVIEW

File: src/utils/args.js [hunk 1]
```diff
@@ -89,4 +90,19 @@ const printArgsUsage = ({ options }) => {
     });
 };
 
-module.exports = { parseProcArgs, printArgsUsage };
+const parseConfigArgs = (configFile ) => {
+  const PATH_PREFIX = '../../'
+  try{
+    // load json file using require
+    const {input, output, lang} = require(path.join(PATH_PREFIX, configFile));
```
Comment (dadolhay) [inline src/utils/args.js:97]: - The file pat's location should be relative to the working directory. please try to use `path.resolve(configFile)`

- Requiring a file executes it if it is not a json, but a js file there.
- Please check if the file exists
- Please read the file onto a variable and use JSON.parse instead, that should be safe

File: ssg.js [hunk 2]
```diff
@@ -39,16 +39,21 @@ if (config.parsedParams.help) {
...
+  language = config.parsedParams.language || language;
+  outputDirPath = config.parsedParams.output || outputDirPath;
+} else {
+  console.log('Input or config parameter missing');
   process.exit(1);
 }
 
-const { language = 'en-CA', output: outputDirPath = DEFAULT_RESULT_FOLDER } =
-  config.parsedParams;
-
```
Comment (dadolhay) [inline ssg.js:51]: This way we'll have to manually maintain each parameter here later.

```suggestion
let { parsedParams } = config;
if (parsedParams.config) {
  config = {
    ...config,
    ...parseConfigArgs(config.parsedParams.config).parsedParams
  };
}

/ Verify input is given
if (!config.parsedParams.input) {
  // eslint-disable-next-line no-console
  console.log('Input parameter missing');
  process.exit(1);
}

const { language = 'en-CA', output: outputDirPath = DEFAULT_RESULT_FOLDER } =
  config;
```

This should automatically merge everything together later on

**Review by dadolhay — CHANGES_REQUESTED** [review]
Also, please add the `prettier` and `eslint` extensions to your VsCode, some of the formatting you did differs from the code style defined in those configs. If you add those, it should be completely automated to format code in the repo.

File: src/utils/args.js [hunk 3]
```diff
@@ -89,4 +91,32 @@ const printArgsUsage = ({ options }) => {
...
+    const data = fs.readFileSync('./src/config/default_config.json', 'utf8');
+    // stringfy file content to object
+    const { input, output, lang } = JSON.stringify(data);
+    return {
+      parsedParams: {
+        input,
+        output,
+        ...(lang && { language: lang }),
+      },
+    };
```
Comment (dadolhay) [inline src/utils/args.js:112]:
```suggestion
    const data = fs.readFileSync(configFile, 'utf8');
    
    return { parsedParams: JSON.parse(data) };
```

File: src/config/default_config.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "input": "./inputs/Sherlock-Holmes-Selected-Stories/testMarkdown.md",
+  "output": "./build",
+  "stylesheet": "https://cdn.jsdelivr.net/npm/water.css@2/out/water.css",
+  "lang": "fr"
```
Comment (dadolhay) [inline src/config/default_config.json:5]:
```suggestion
  "language": "fr"
```

File: ssg.js [hunk 5]
```diff
@@ -39,16 +43,22 @@ if (config.parsedParams.help) {
...
+  outputDirPath = config.parsedParams.output || outputDirPath;
+} else {
+  // eslint-disable-next-line
+  console.log('Input or config parameter missing');
   process.exit(1);
 }
 
-const { language = 'en-CA', output: outputDirPath = DEFAULT_RESULT_FOLDER } =
-  config.parsedParams;
-
```
Comment (dadolhay) [inline ssg.js:51]: Your solution would require us fo modify this merging code for every new config param introduced. Please use a solution like this instead

```suggestion
let { parsedParams } = config;
if (parsedParams.config) {
  parsedParams = {
    ...parsedParams,
    ...parseConfigArgs(config.parsedParams.config).parsedParams
  };
}
/ Verify input is given
if (!parsedParams.input) {
  // eslint-disable-next-line no-console
  console.log('Input parameter missing');
  process.exit(1);
}
const { language = 'en-CA', output: outputDirPath = DEFAULT_RESULT_FOLDER } =
  parsedParams;
```

**Review by dadolhay — CHANGES_REQUESTED** [review]

File: ssg.js (hunk 5 above)
Comment (IvaniGabrovsky) [reply ssg.js:51]: Hi Denes, the following recomendation file is failing when I run with -config. I am getting error:  
No txt files to process

Comment (dadolhay) [reply ssg.js:51]: If you also change 
`inputPath: config.parsedParams.input,` to `inputPath: parsedParams.input,`
in line `67`, it should work (I tested it)

Comment (IvaniGabrovsky) [reply ssg.js]: I fixed it. Please approve the PR and merge.

**Comment by IvaniGabrovsky** [comment]
Hi Denise this is fixed in this commit 9a40c3e.

**Review by dadolhay — APPROVED** [review]

**Pull request merged**